src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -248,21 +249,31 @@ public SyntaxTriviaList InsertRange(int index, IEnumerable<SyntaxTrivia> trivia)
                 throw new ArgumentOutOfRangeException(nameof(index));
             }
 
-            var items = trivia.ToList();
-            if (items.Count == 0)
+            if (trivia == null)
             {
-                return this;
+                throw new ArgumentNullException(nameof(trivia));
             }
 
-            var list = this.ToList();
-            list.InsertRange(index, items);
+            var builder = SyntaxTriviaListBuilder.Create();
+
+            var current = 0;
+            foreach (var thisTrivia in this)
```

Comment:
>foreach (var thisTrivia in this) [](start = 12, length = 32)

Is there a perf benefit in using foreach and keeping track of position at the same time? Please add a comment about this. Otherwise, please use two for loops instead.